build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/RenameTestCases.cs

```diff
@@ -134,28 +144,49 @@ static void CreateErrorResultsFile (string source, string dest, Exception e)
 						new XAttribute ("current-uiculture", "en-US")),
 					new XElement ("test-suite",
 						new XAttribute ("type", "APK-File"),
-						new XAttribute ("name", dest),
+						new XAttribute ("name", testSuiteName),
 						new XAttribute ("executed", "True"),
 						new XAttribute ("result", "Failure"),
 						new XAttribute ("success", "False"),
 						new XAttribute ("time", "0"),
 						new XAttribute ("asserts", "0"),
 						new XElement ("results",
 							new XElement ("test-case",
-								new XAttribute ("name", Path.GetFileName (dest)),
+								new XAttribute ("name", testCaseName),
 								new XAttribute ("executed", "True"),
 								new XAttribute ("result", "Error"),
 								new XAttribute ("success", "False"),
 								new XAttribute ("time", "0.0"),
 								new XAttribute ("asserts", "1"),
 								new XElement ("failure",
 									new XElement ("message",
-										$"Error processing `{source}`.  " +
+										$"Error processing `{sourceFile}`.  " +
 										$"Check the build log for execution errors.{Environment.NewLine}" +
 										$"File contents:{Environment.NewLine}",
-										new XCData (contents)),
+										new XCData (contents.ToString ())),
 									new XElement ("stack-trace", e.ToString ())))))));
-			doc.Save (dest);
+			doc.Save (destFile);
+		}
+
+		// Example `SourceFile`:
+		//   /Users/builder/jenkins/workspace/xamarin-android-pr-builder-release/xamarin-android/bin/TestRelease/TestResult-Mono.Android_Tests.xml
+		// Example `DestinationFolder`:
+		//   /Users/builder/jenkins/workspace/xamarin-android-pr-builder-release/xamarin-android/
+		// Example `adb logcat`:
+		//   /Users/builder/jenkins/workspace/xamarin-android-pr-builder-release/xamarin-android/tests/logcat-Release-Mono.Android_Tests.txt
+		//
+		// We need to extract the "base" test name from `SourceFile`, and use that to construct `logcatPath`
+		static void GetTestCaseInfo (string sourceFile, string destinationFolder, string config, out string testSuiteName, out string testCaseName, out string logcatPath)
+		{
+			var name        = Path.GetFileNameWithoutExtension (sourceFile);
+			if (name.StartsWith ("TestResult-"))
+				name    = name.Substring ("TestResult-".Length);
```

Comment:
I think we should also cut the ending `-{configuration}` here. So from `TestResult-Mono.Android_Tests-Debug.xml` get just `Mono.Android_Tests`, right?

Reply:
`name` won't contain `-{config}`.  See [other comment](
https://github.com/xamarin/xamarin-android/pull/2550#discussion_r242911867).